Framework/Muon/src/MuonAlgorithmHelper.cpp

```diff
@@ -220,6 +220,96 @@ void groupWorkspaces(const std::string &groupName,
   }
 }
 
+/**
+ * Generate a string (to be included in workspace names) to signify the
+ * period algebra.
+ * Format: a sorted sequence of integers separated by only "+,-" in the form
+ * "<summed>-<sub>" with <summed>, <sub> e.g. 1+2+3.
+ * @param summedPeriods :: [input] comma separated string of integers.
+ * @param subtractedPeriods :: [input] comma separated string of integers.
+ * @returns :: string for use in workspace name.
+ */
+std::string generatePeriodAlgebraString(std::string summedPeriods,
+                                        std::string subtractedPeriods) {
+
+  if (summedPeriods.empty()) {
+    return "";
+  }
+  // the default for single period data
+  if (summedPeriods == "1" && subtractedPeriods == "") {
+    return "";
+  }
+
+  std::vector<int> summedPeriodVec, subtractedPeriodVec;
+  const char delimeter(',');
+
+  std::vector<std::string> summedPeriodStrVec;
+  boost::split(summedPeriodStrVec, summedPeriods,
+               [delimeter](char c) { return c == delimeter; });
+  if (!summedPeriodStrVec[0].empty()) {
+    transform(summedPeriodStrVec.begin(), summedPeriodStrVec.end(),
+              back_inserter(summedPeriodVec), [](const std::string &p) {
+                if (std::all_of(p.begin(), p.end(), ::isdigit)) {
+                  return std::stoi(p);
+                } else {
+                  throw std::invalid_argument(
+                      "List contains a non-integer value : " + p);
+                }
+              });
+  }
+
+  std::vector<std::string> subtractedPeriodStrVec;
+  boost::split(subtractedPeriodStrVec, subtractedPeriods,
```

Comment:
as far as I can tell this is the same as the above except for the vector it populates and the vector that it process. Could we replace both with a function call?